modules/wordpress-com-block-editor-iframe.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * Module Name: WordPress.com Block Editor Iframe
+ * Module Description: Allow new block editor posts to be composed on WordPress.com.
+ * Jumpstart Description: Allow new block editor posts to be composed on WordPress.com.
+ * Sort Order: 15
+ * First Introduced: 7.3
+ * Requires Connection: Yes
+ * Auto Activate: Yes
+ * Module Tags: Writing
+ * Feature: Writing
+ * Additional Search Queries: iframes, allow, compose, WordPress.com, block, editor, post
+ */
+
+function jetpack_disable_send_frame_options_header() {
+	if ( jetpack_framing_allowed() ) {
+		remove_action( 'admin_init', 'send_frame_options_header' );
+	}
+}
+add_action( 'admin_init', 'jetpack_disable_send_frame_options_header', 1 ); // High priority to get ahead of send_frame_options_header
+
+function jetpack_framing_allowed() {
+	if ( ! empty( $_GET['frame-nonce'] ) && false !== strpos( $_GET['frame-nonce'], '.' ) ) {
+		list( $token, $signature ) = explode( '.', $_GET['frame-nonce'] );
+
+		$verified = Jetpack::init()->verify_xml_rpc_signature( $token, $signature );
+
+		if ( $verified ) {
+			if ( ! defined( 'IFRAME_REQUEST' ) ) {
+				define( 'IFRAME_REQUEST', true );
+			}
+
+			return true;
+		}
+	}
+
+	return false;
```

Comment:
it might help to invert our conditional here and perform an early-abort

```php
function jetpack_framing_allowed() {
	if ( empty( $_GET['frame-nonce'] ) || false === strpos( $_GET['frame-nonce'], '.' ) ) {
		return false;
	}

	list( $token, $signature ) = explode( '.', $_GET['frame-nonce'] );

	…

	if ( $verified && ! defined( 'IFRAME_REQUEST' ) ) {
		define( 'IFRAME_REQUEST', true );
	}

	return (bool) $verified;
}
```

Reply:
Oh yes, much nicer – fixed in c4fe9bfb1f56b5319709447db4ab222905ae5751.